src/cmd/hotspots/transfer.rs

```diff
@@ -0,0 +1,150 @@
+        OutputFormat,
+    },
+    keypair::PubKeyBin,
+    result::Result,
+    traits::{Sign, ToJson, TxnEnvelope, TxnFee, B58, B64},
+};
+use helium_api::{
+    BlockchainTxn, BlockchainTxnTransferHotspotV1, Client, Hnt, PendingTxnStatus, Txn,
+};
+use std::io;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+/// Transfer hotspot as buyer or seller.
+pub enum Transfer {
+    /// Create and sign transaction to sell a hotspot, outputting it as base64 for counter-party
+    Sell(Sell),
+    /// Ingest a transaction to buy a hotspot from base64.
+    /// Signs and submits the transaction to the API
+    Buy(Buy),
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Sell {
+    /// Public address of gateway to be transferred
+    gateway: String,
+    /// The recipient of the gateway transfer
+    buyer: String,
+    /// Price in HNT to be paid by recipient of transfer
+    price: Option<Hnt>,
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Buy {
+    /// Base64 encoded transaction to sign. If no transaction if given
+    /// stdin is read for the transaction. Note that the stdin feature
+    /// only works if the wallet password is set in the
+    /// HELIUM_WALLET_PASSWORD environment variable
+    #[structopt(name = "TRANSACTION")]
+    txn: Option<String>,
+    #[structopt(long)]
+    /// Commit to sign and submit the transaction
+    commit: bool,
+}
+
+impl Buy {
+    fn read_txn(&self) -> Result<String> {
+        match &self.txn {
+            Some(txn) => Ok(txn.to_string()),
+            None => {
+                let mut buffer = String::new();
+                io::stdin().read_line(&mut buffer)?;
+                Ok(buffer.trim().to_string())
+            }
+        }
+    }
+}
+
+impl Transfer {
+    pub fn run(self, opts: Opts) -> Result {
+        let wallet = load_wallet(opts.files)?;
+        let client = Client::new_with_base_url(api_url());
+
+        match self {
+            Self::Sell(sell) => {
+                let seller = wallet.pubkey_bin.to_vec();
+                let buyer = PubKeyBin::from_b58(&sell.buyer)?;
+                let buyer_account = client.get_account(&buyer.to_b58()?)?;
+                let gateway = PubKeyBin::from_b58(&sell.gateway)?.to_vec();
+
+                let mut txn = BlockchainTxnTransferHotspotV1 {
+                    fee: 0,
+                    seller,
+                    gateway,
+                    buyer: buyer.to_vec(),
+                    seller_signature: vec![],
+                    buyer_signature: vec![],
+                    amount_to_seller: if let Some(price) = sell.price {
+                        price.to_bones()
+                    } else {
+                        0
+                    },
+                    buyer_nonce: buyer_account.speculative_nonce + 1,
+                };
+                txn.fee = txn.txn_fee(&get_txn_fees(&client)?)?;
+                eprintln!("{:#?}", txn.to_json()?);
```

Comment:
What is this doing here?

Reply:
Switched this to use the print_json. I think it's worth outputting the JSON of the transaction for user confirmation? Can remove if you prefer

Reply:
Hmm.. I've been trying to avoid doing _two_ prints of transactions. What do the other commands do?

Reply:
I don't believe there's a precedent for outputting base64 in the wallet. But if we take the miner UX, it only prints out the transaction in base64 and does not confirm it in JSON. So we can drop this JSON output to be consistent with that UX.

eg:
```
$ miner txn add_gateway owner=13aCNAYg2YL6fJCCYJqMNPKVvBp3fVKSWoKckvzyePSPS95iwDh --payer 14fzfjFcHpDR1rTH8BNPvSi5dKBbgxaDnmsVPbCjuq9ENjpZbxh
CrwBCiEBUsSEnLZRRLpls0Fw9yo6VrskNBXI5MgVZlFeyFMRCEYSIQAv1aPyg0ym4jfrrndsKgeirFM+VmEJNw6OmBIihLQadyJIMEYCIQCIxNCah47SQX7+z2A9Am1hYxTVzAyeFpsDcuqNDewtvwIhAOSEUpP7ujyTkbBTvuegRueKdQhd7bGg8faQJX0d5byTKiEB46EqIVg62+pC1Vx15S85zTV+N2Y4z6Caz+14sFxpso04gJL0AUDo+wM=
```